studio/src/pages/RequestDetailsPage/hooks/useReplayRequest.ts

```diff
@@ -0,0 +1,159 @@
+import { useMakeProxiedRequest } from "@/pages/RequestorPage/queries";
+import { useRequestor } from "@/pages/RequestorPage/reducer";
+import type { OtelSpan } from "@/queries";
+import { useCallback, useMemo } from "react";
+import {
+  getRequestBody,
+  getRequestHeaders,
+  getRequestMethod,
+  getRequestQueryParams,
+  getRequestUrl,
+} from "../v2/otel-helpers";
+
+export function useReplayRequest({ span }: { span?: OtelSpan }) {
+  if (!span) {
+    return {
+      replay: () => {},
+      isReplaying: false,
+    };
+  }
+  const method = getRequestMethod(span);
+
+  const pathWithSearch = useMemo<string>(() => {
+    return getRequestUrl(span);
+  }, [span]);
+
+  const url = new URL(pathWithSearch);
+  const replayBaseUrl = url.origin;
+  const replayPath = url.pathname;
+
+  const requestHeaders = useMemo<Record<string, string>>(() => {
+    return getRequestHeaders(span) ?? {};
+  }, [span]);
+
+  const filterReplayHeaders = useCallback(
+    (
+      headers: Array<{
+        key: string;
+        value: string;
+        id: string;
+        enabled: boolean;
+      }>,
+    ) => {
+      const excludedHeaders = [
+        "x-fpx-trace-id",
+        "transfer-encoding",
+        "content-length",
+        "host",
+        "connection",
+        "accept-encoding",
+        "user-agent",
+        "referer",
+        "origin",
+        "cookie",
+        "sec-fetch-site",
+        "sec-fetch-mode",
+        "sec-fetch-dest",
+        "sec-ch-ua",
+        "sec-ch-ua-mobile",
+        "sec-ch-ua-platform",
+        "pragma",
+        "cache-control",
+        "te",
+      ];
+
+      return headers.filter(
+        (header) => !excludedHeaders.includes(header.key.toLowerCase()),
+      );
+    },
+    [],
+  );
+
+  const replayHeaders = useMemo(() => {
+    const headers = Object.entries(requestHeaders).map(([key, value]) => ({
+      id: key,
+      key,
+      value,
+      enabled: true,
+    }));
+
+    return filterReplayHeaders(headers);
+  }, [requestHeaders, filterReplayHeaders]);
+
+  const replayBody = useMemo(() => {
+    const body = getRequestBody(span);
+    try {
```

Comment:
no need to refactor, but for future reference, i made a utility for this kind of logic

utility is called `isJson`, should be importable from the utils module